t/14-grutasks.t

```diff
@@ -647,12 +647,18 @@ subtest 'handling dying GRU task' => sub {
 subtest 'git clone' => sub {
     my $openqa_utils = Test::MockModule->new('OpenQA::Task::Git::Clone');
     my @mocked_git_calls;
+    my $clone_dirs = {
+        '/etc/' => 'http://localhost/foo.git',
+        '/root/' => 'http://localhost/foo.git#foobranch',
+        '/this_directory_does_not_exist/' => 'http://localhost/bar.git',
+    };
     $openqa_utils->redefine(
         run_cmd_with_log_return_error => sub ($cmd) {
             my $stdout = '';
             $stdout = 'ref: refs/heads/master	HEAD' if $cmd->[3] eq 'ls-remote';
-            $stdout = 'http://localhost/foo.git' if $cmd->[4] eq 'get-url';
+            $stdout = $clone_dirs->{$cmd->[2]} =~ s/#.*//r if $cmd->[4] eq 'get-url';
```

Comment:
How about making this a little more explicit to avoid having to trace what the numbers mean?

```suggestion
            my dir = $cmd->[2];
            my $action = $cmd->[4];
            $stdout = $clone_dirs->{$dir} =~ s/#.*//r if $action eq 'get-url';
```

Reply:
One problem is that the dir is not always item number 2, and action is not always number 4. When we call the `env SSH... git ...` wrapper the numbers are different.

Reply:
But then how does this work in the first place? The code applies to all of the commands, no?

Reply:
Maybe it would make sense to split the hash into something like label/command/action?

Reply:
> But then how does this work in the first place? The code applies to all of the commands, no?

Yep.
When the action is `ls-remote`, it's on position 3, and for `get-url` it's on position 4.

Reply:
> Maybe it would make sense to split the hash into something like label/command/action?

I can't imagine right now which hash you mean and how it would look like? Maybe a short example?

Reply:
I changed it a bit. I hope it's clearer now.

Reply:
Something like this:

```
    my $expected_calls = [
        # /etc/
        [action => 'get-url', cmd => 'git -C /etc/ remote get-url origin'],
        [action => 'ls-remote', cmd => 'env GIT_SSH_COMMAND="ssh -oBatchMode=yes" git ls-remote --symref http://localhost/foo.git HEAD'],
```